src/main/java/fi/hsl/transitdata/cancellation/util/BulletinUtils.java

```diff
@@ -14,8 +19,67 @@ public static List<InternalMessages.Bulletin> filterMassCancellationsFromBulleti
                                         bulletin.getPriority() == InternalMessages.Bulletin.Priority.WARNING)
                 .collect(Collectors.toList());
     }
-
-    public static boolean bulletinAffectsAll(InternalMessages.Bulletin bulletin) {
-        return bulletin.getAffectsAllRoutes() || bulletin.getAffectsAllStops();
+    
+    public static List<CancellationData> parseCancellationDataFromBulletins(List<InternalMessages.Bulletin> massCancellations, String digitransitDeveloperApiUri) {
+        List<CancellationData> tripCancellations = massCancellations.stream().
+                flatMap(mc -> createTripCancellations(mc, digitransitDeveloperApiUri).stream()).collect(Collectors.toList());
+        return tripCancellations;
+    }
+    
+    // One cancellation contains one trip
+    // A route consists of many trips
+    private static List<CancellationData> createTripCancellations(InternalMessages.Bulletin massCancellation, String digitransitDeveloperApiUri) {
+        // TODO: this implementation is not final
+        List<CancellationData> tripCancellations = new ArrayList<>();
+        
+        LocalDateTime validFrom = Instant.ofEpochMilli(
+                massCancellation.getValidFromUtcMs()).atZone(ZoneId.systemDefault()).toLocalDateTime();
+        
+        LocalDateTime validTo = Instant.ofEpochMilli(
+                massCancellation.getValidToUtcMs()).atZone(ZoneId.systemDefault()).toLocalDateTime();
```

Comment:
Apparently UTC -> Timezone conversions.

This would result in dates being incorrect!

Reply:
Actually this is probably only misleading method name. Will add comment on necessary bit also...

Reply:
I don't get it, can you fix this?

Reply:
This is non-issue. Thinking with fresh brains even the name kinda makes sense